scylla/src/transport/session.rs

```diff
@@ -1862,9 +1868,15 @@ impl Session {
     }
 
     pub async fn check_schema_agreement(&self) -> Result<bool, QueryError> {
-        let connections = self.cluster.get_working_connections()?;
+        let cluster_data = self.get_cluster_data();
+        let connections_iter = cluster_data.iter_working_connections();
 
-        let handles = connections.iter().map(|c| c.fetch_schema_version());
+        let handles = connections_iter.map(|c| async {
+            match c {
+                Ok(c) => c.fetch_schema_version().await,
+                Err(err) => Err(err),
+            }
+        });
```

Comment:
You probably mean "instead of `match`". Then I don't think so, because the operation is done inside an `async` block, which is a known limitation of functional idioms in Rust.